src/jsonSchemaTranslator.ts

```diff
@@ -142,6 +142,9 @@ export async function updateSchema(meta: ISourceMeta, ch: ClickhouseConnection)
   await Promise.all(meta.children.map((child) => updateSchema(child, ch)))
 
   const isRoot = meta.pkMappings.filter((pkMap) => pkMap.pkType === PKType.ROOT).length == 0
+  if (!(await ch.listTables()).includes(meta.sqlTableName.slice(1, meta.sqlTableName.length - 1))) {
```

Comment:
To avoid making the same call several times, we could use the list `existingTables` available in `createStreamProcessor`